sdk/python/kfp/components/for_loop.py

```diff
@@ -272,3 +272,45 @@ def _get_name_override(self, loop_arg_name: str, subvar_name: str) -> str:
             The name of this loop arg variable.
         """
         return f'{loop_arg_name}{self.SUBVAR_NAME_DELIMITER}{subvar_name}'
+
+
+class Collected(pipeline_channel.PipelineChannel):
+    """For collecting outputs from dsl.ParallelFor.
+
+    Args:
+        output: The output of an upstream task within a dsl.ParallelFor sub-DAG.
```

Comment:
Note that this is forward compatible with allowing >1 output to be passed by changing `__init__` to

```python
def __init__(self, output, *args):
    self.outputs = [output] + list(args)
```